src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.x < args.pointA.x)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::south);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+    }
+
+    // 0x00462DCE
+    static uint32_t lowerRaiseLandMountain(const LowerRaiseLandMountainArgs& args, const uint8_t flags)
+    {
+        mtnToolXParams = (args.pointB.x << 16) | args.pointA.x;
+        mtnToolYParams = (args.pointB.y << 16) | args.pointA.y;
+        mtnToolGCFlags = flags;
+
+        if (flags & Flags::apply)
+        {
+            S5::getOptions().madeAnyChanges = 1;
+        }
+
+        mtnToolCost = 0;
+
+        // We keep track of removed buildings for each tile visited
+        // this prevents accidentally double counting their removal
+        // cost if they span across multiple tiles.
+        std::set<World::Pos3, LessThanPos3> removedBuildings{};
+
+        // Play sound if this is a company action
+        if ((flags & Flags::apply) && getCommandNestLevel() == 1 && getUpdatingCompanyId() != CompanyId::neutral)
+        {
+            const auto height = TileManager::getHeight(args.centre).landHeight;
+            Audio::playSound(Audio::SoundId::construct, World::Pos3(args.centre.x, args.centre.y, height));
+        }
+
+        // First, raise/lower the mountain's centre tile
+        {
+            auto result = adjustMountainCentre(args, removedBuildings, flags);
+            if (result != FAILURE)
+            {
+                mtnToolCost = *mtnToolCost + result;
+            }
+        }
+
+        if (!validCoords(args.pointA) || !validCoords(args.pointB))
+        {
+            return FAILURE;
+        }
+
+        // 0x00462E7E
+        auto preTile = TileManager::get(args.pointA);
+        auto* preSurface = preTile.surface();
+        if (preSurface->slope() != 0)
+        {
+            auto result = adjustMountainCentre(args, removedBuildings, flags);
+            if (result != FAILURE)
+            {
+                mtnToolCost = *mtnToolCost + result;
+            }
+
+            // Verify that the slope is now gone. Abort if not.
+            auto tile = TileManager::get(args.pointA);
+            auto* surface = tile.surface();
+            if (surface->slope() != 0)
+            {
+                // 0x004633CB
+                GameCommands::setExpenditureType(ExpenditureType::Construction);
+
+                auto tileHeight = World::TileManager::getHeight(args.centre);
+                GameCommands::setPosition(World::Pos3(args.centre.x, args.centre.y, tileHeight.landHeight));
+
+                return mtnToolCost;
+            }
+        }
+
+        // 0x00462F25
+        const int16_t minRadius = ((args.pointB.x - args.pointA.x) >> 5) - 1;
+        const int16_t maxRadius = minRadius + 64;
```

Comment:
When working with world coordinates its probably best to use coord_t which will simplify the switch from 16 to 32 all at once at some point.

Reply:
Doesn't really feel like a coordinate type is appropriate for a radius though, right? I can switch to `auto` or `int32_t` to facilitate the transition in the future, though? (Not that these radii should ever be bigger than an `int8_t`, but still...)

Reply:
Why not? The radius defines how many units in the world space.

Reply:
Also given we use signed types we should also avoid the shift and just divide it by tile size, I guess the original uses shr which is unsigned shift but using shift on signed types in cpp will not use the shr instruction instead it will most likely become sar.

Reply:
Sure, I'll use `tile_coord_t` then.